File: frontend/src/types/projectAlertWebhook.ts
export interface ProjectAlertWebhook {
  project_id: number;
  enabled: boolean;
  url: string;
  last_delivery_attempted_at: string | null;
  last_delivery_transition: string | null;
  last_delivery_outcome: "delivered" | "failed" | null;
  last_delivery_http_status: number | null;
  last_delivery_error: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface ProjectAlertWebhookUpdateInput {
  enabled: boolean;
  url: string;
}

File: backend/app/schemas/project_alert_webhook.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class ProjectAlertWebhookUpdate(BaseModel):
    enabled: bool
    url: str = Field(default="", max_length=2048)


class ProjectAlertWebhookRead(BaseModel):
    project_id: int
    enabled: bool
    url: str
    last_delivery_attempted_at: datetime | None
    last_delivery_transition: str | None
    last_delivery_outcome: Literal["delivered", "failed"] | None
    last_delivery_http_status: int | None
    last_delivery_error: str | None
    created_at: datetime | None
    updated_at: datetime | None

    model_config = ConfigDict(from_attributes=True)
